src/app/backing/components/AllocationBar/AllocationBar.tsx

```diff
@@ -76,37 +104,51 @@ const AllocationBar: React.FC<AllocationBarProps> = ({
       window.removeEventListener('mousemove', handleResize)
       window.removeEventListener('mouseup', onMouseUp)
     }
-    // eslint-disable-next-line
-  }, [dragIndex, values])
+  }, [dragIndex, currentValues])
 
-  // dnd-kit sort logic
-  function handleDragEnd({ active, over }: DragEndEvent) {
+  // dnd-kit reorder logic
+  const handleDragEnd = ({ active, over }: DragEndEvent) => {
     if (!over || active.id === over.id) return
-    const oldIndex = itemsData.findIndex(item => item.key === active.id)
-    const newIndex = itemsData.findIndex(item => item.key === over.id)
-    setItemsData(arrayMove(itemsData, oldIndex, newIndex))
-    setValues(arrayMove(values, oldIndex, newIndex))
+
+    const oldIndex = currentItems.findIndex(item => item.key === active.id)
+    const newIndex = currentItems.findIndex(item => item.key === over.id)
+
+    const newItems = arrayMove(currentItems, oldIndex, newIndex)
+    const newValues = arrayMove(currentValues, oldIndex, newIndex)
+
+    if (isControlled) {
+      onChange?.({
```

Comment:
What's the purpose of this? when the items are sorted, the values don't change, so I don't get why we should raise it to the parent (causing another render)